flowmachine/flowmachine/core/custom_query.py

```diff
@@ -33,13 +37,26 @@ class CustomQuery(Query):
     .table.Table for an equivalent that deals with simple table access
     """
 
-    def __init__(self, sql):
+    def __init__(self, sql: str, column_names: List[str]):
         """
 
         """
 
         self.sql = pretty_sql(sql)
+        self._column_names = column_names
         super().__init__()
 
+    @property
+    def column_names(self) -> List[str]:
+        return self._column_names
+
     def _make_query(self):
         return self.sql
+
+    def __getstate__(self):
```

Comment:
I'm not sure I understand what's going on in this method. First of all, why would it try to delete the attribute `_column_names`? Is this not essential to determine this query? Secondly, why would the `del` line ever fail, since this query does have an attribute called `self._column_names`? Would be good to explain these things in a docstring or comment.